### DISCUSSION TITLE
Electromagnetics Module Based on edge element or node element？

### DISCUSSION THREAD

**Discussion by AdelineHunter** [idaholab/moose#21673] [Q&A Modules: Electromagnetics] [answered]
Hi,

Is the Electromagnetics Module developed based on the edge element method? Or based on the node element method？

Thanks

**Comment by cticenhour** [comment, marked as answer]
If you choose to solve the vector fields component-wise, you can use node elements. If you solve them directly as vectors, you use edge elements (specifically, Nedelec first order). See the benchmark examples for general module usage, and most of those use the edge element.

**Reply by cticenhour** [reply]
One more note - you could use LAGRANGE_VEC type elements. That's just a vector of Lagrange node elements. Can be useful for transitioning an input file from component-wise to full vector.
